src/shared/dissect-image.c

```diff
@@ -1268,12 +1314,13 @@ static int mount_partition(
         assert(m);
         assert(where);
 
+        if (!m->found || m->mount_node_fd < 0)
```

Comment:
the left-hand check is redundant, no? the right-hand check already should be enough?